### DISCUSSION TITLE
Manual/shader-shaderlab-commands

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#421] [Notes]
# Manual/shader-shaderlab-commands

https://docs.unity3d.com/Manual/shader-shaderlab-commands.html

**Comment by vertxxyz** [comment]
If you've declared properties to set these values, say:
```csharp
[HideInInspector] _ZTest("__zt", Float) = 4.0
...
ZTest [_ZTest]
```
The possible values for all these from C# are as below:
| Key | Value | Syntax |
|------|-----|-----|
| AlphaToMask | `0, 1` | `AlphaToMask[_AlphaToMask]` |
| Blend | [`BlendMode`](https://docs.unity3d.com/ScriptReference/Rendering.BlendMode.html) | `Blend [_SrcBlend] [_DstBlend]`,<br>`Blend [_SrcBlend][_DstBlend], [_SrcBlendAlpha][_DstBlendAlpha]` |
| BlendOp | [`BlendOp`](https://docs.unity3d.com/ScriptReference/Rendering.BlendOp.html) | `BlendOp [_BlendOp]` |
| ColorMask | [`ColorWriteMask`](https://docs.unity3d.com/ScriptReference/Rendering.ColorWriteMask.html) | `ColorMask [_ColorMask]` |
| Conservative | `0, 1` |  `Conservative [_Conservative]` |
| Cull | [`CullMode`](https://docs.unity3d.com/ScriptReference/Rendering.CullMode.html) | `Cull [_CullMode]` |
| Offset | `0->1` | `Offset [_ZBias], [_ZBias]` |
| Stencil | See [Stencil](https://docs.unity3d.com/Manual/SL-Stencil.html) | See [Stencil](https://docs.unity3d.com/Manual/SL-Stencil.html) |
| ZClip | `0, 1` | `ZClip [_ZClip]` | 
| ZTest | [`CompareFunction`](https://docs.unity3d.com/ScriptReference/Rendering.CompareFunction.html) | `ZTest [_ZTest]` |
| ZWrite | `0, 1` | `ZWrite [_ZWrite]` |
